### PR TITLE
[pipelining] Test schedules against manual stage

### PR DESCRIPTION
**Pull request by kwen2501**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #126735
* #126721
* #126812

Added manual stage in test_schedule.py so that we can test various schedules against it.

In this file we now have:
- test_schedule_with_tracer
- test_schedule_with_manual
- test_grad_with_tracer
- test_grad_with_manual

Tested schedules are:
- ScheduleGPipe
- Schedule1F1B

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k

### CODE REVIEW

File: test/distributed/pipelining/test_schedule.py [hunk 1]
```diff
@@ -100,8 +101,51 @@ def test_ec_backward(self, ScheduleClass):
     @requires_nccl()
     @skip_but_pass_in_sandcastle_if(not TEST_MULTIGPU, "NCCL test requires 2+ GPUs")
     @parametrize("ScheduleClass", [ScheduleGPipe, Schedule1F1B])
-    def test_grad(self, ScheduleClass):
-        mod = MultiMLP(d_hid)
+    def test_schedule_with_manual(self, ScheduleClass):
```
Comment (wconstab) [inline test/distributed/pipelining/test_schedule.py:104]: just curious- is there anything that test_schedule* tests that test_grad* does not test? or should we just delete test_schedule and rename test_grad to test_schedule?

**Review by wconstab — APPROVED** [review]

**Review by H-Huang — APPROVED** [review]
LGTM!

File: test/distributed/pipelining/test_schedule.py (hunk 1 above)
Comment (kwen2501) [reply test/distributed/pipelining/test_schedule.py:104]: Makes sense, maybe I can consolidate them.

**Pull request closed without merging**